tools/util/include/cutlass/util/device_memory.h

```diff
@@ -58,11 +58,14 @@ T* allocate(size_t count = 1) {
 
   bytes = count * sizeof(T);
 
+#if defined(CUTLASS_ENABLE_SYCL)
+  ptr = reinterpret_cast<T*>(syclcompat::malloc(bytes));
```

Comment:
@muhammad-tanvir-1211 , I believe this needs to be `sycl::aligned_alloc_device`(something which we might need to add in syclcompat) and not simply just a `malloc_device`.  cudaMalloc always returns 256 Byte aligned memory, and hence they are able to do vector loads from global memory without worrying about alignment. `malloc_device` makes no such guarantees. Our sycl path in cuda works because the underlying driver API which is called will always return aligned memory anyways...

However for intel we might need to do this, as the 2D block copies require 64 Byte alignment as well,

Reply:
Also, a null pointer check in case of failed allocation